docs/syz_verifier.md

```diff
@@ -0,0 +1,93 @@
+# syz-verifier
+
+Testing and fuzzing aim to verify a subset of the behaviours of running programs: any logical error can cause
+bugs that make the program behave incorrectly, possibly producing unintended output. Many bugs, however, do
+not cause the program to crash nor produce assertion failures.
+
+The behaviour of a program can be classified as either defined or undefined. Undefined behaviour can easily be detected by either static (e.g. compiler warnings) or dynamic analysis (e.g. sanitizers). However, logical errors that result in defined program behaviour, yet are still causing
+bugs that make the program behave incorrectly (according to some documented or otherwise expected intent),
+are much harder to detect. One way to detect such bugs is setting up test cases that compare program outputs
+against a set of expected results (e.g. by manually creating integration or unit tests).
+
+Differential fuzzing automates this process by providing the same input to different implementations and then
+cross-comparing the resulting behaviour to determine whether it is identical.
```

Comment:
append:
```
Here it is assumed that one of the implementations is a correct reference implementation. For example, such a reference implementation could be an identically behaving model, but otherwise very different implementation; but it could also be an older assumed correct version of the same implementation.
```